std/experimental/allocator/package.d

```diff
@@ -922,9 +922,26 @@ private void fillWithMemcpy(T)(void[] array, auto ref T filler) nothrow
 
 private T[] uninitializedFillDefault(T)(T[] array) nothrow
 {
-    T t = T.init;
-    fillWithMemcpy(array, t);
-    return array;
+    static if (isInitAllZeroBits!T)
+    {
+        import core.stdc.string : memset;
+        if (array !is null)
+            memset(array.ptr, 0, T.sizeof * array.length);
+        return array;
+    }
+    else static if (is(T : char) || (is(T : wchar) && T.init == 0xffff))
```

Comment:
1) What's the reason for excluding `dchar` here?

2) Shouldn't the type classifiers be removed? E.g.
- [`Unqual`]((https://dlang.org/phobos/std_traits.html#Unqual)
- [isSomeChar](https://dlang.org/phobos/std_traits.html#isSomeChar) - doesn't allow `alias this`

3) Also the `T.init == 0xffff` check is to support user-extended versions of (char) that don't overwrite the `init` value, right? This should probably get a test too.

Reply:
1. `dchar.init == 0x0000_ffff`. Can't use memset.

2. `static assert(is(immutable(char) : char))` compiles. I don't know that Unqual is necessary. I hadn't given much though to `alias this`. What do you suggest?

3. Since `char.sizeof == 1` memset will work regardless of what char.init is. Since `wchar.sizeof == 2` memset will work if the high 8 bits are the same as the low 8 bits; the main purpose of the `T.init == 0xffff` check was reminding the casual reader that is is true for wchar. Maybe`0 == T.init ^ (T.init >>> 8)` would be better?